src/caffe/solver.cpp

```diff
@@ -860,6 +860,214 @@ void AdaGradSolver<Dtype>::ComputeUpdateValue(int param_id, Dtype rate) {
 }
 
 template <typename Dtype>
+void AdaDeltaSolver<Dtype>::PreSolve() {
+  // Add the extra history entries for AdaDelta after those from
+  // SGDSolver::PreSolve
+  const vector<shared_ptr<Blob<Dtype> > >& net_params = this->net_->params();
+  for (int i = 0; i < net_params.size(); ++i) {
+        const vector<int>& shape = net_params[i]->shape();
+        this->history_.push_back(
+                shared_ptr<Blob<Dtype> >(new Blob<Dtype>(shape)));
+  }
+}
+
+template <typename Dtype>
+void AdaDeltaSolver<Dtype>::Regularize(int param_id) {
+  const vector<shared_ptr<Blob<Dtype> > >& net_params = this->net_->params();
```

Comment:
The only difference between your `AdaDeltaSolver::Regularize` and the original `SGDSolver::Regularize` seem to be that you use `const vector<shared_ptr<Blob<Dtype> > >& net_params` instead of `const vector<Blob<Dtype>*>& net_params`. The rest are all the same.

So, I believe we don't need a `AdaDeltaSolver::Regularize` here. Let's just use `SGDSolver::Regularize` instead.